src/Resources/config/doctrine/AuditLog.orm.xml

```diff
@@ -7,6 +7,8 @@
         <id name="id" type="bigint" column="id">
             <generator strategy="AUTO" />
         </id>
+        <field name="ip" nullable="true" type="string" length="32"/>
+        <field name="userAgent" nullable="true" type="string" length="255"/>
```

Comment:
Set the type to text and remove the length attribute, this should prevent the length from being truncated.

Reply:
Are you sure? Some of those UA strings can get pretty long. I've read many comments in many threads over the past two days that keep reinforcing what a bad idea it is to be logging UAs in the first place. that said, i think if we want to get fully into the business of logging these UAs properly without truncation, then it should probably be normalised out to a joined table, using a fixed length hash of the UA as a unique key on the table in order to avoid duplication of these potentially very long strings
(https://stackoverflow.com/questions/654921/how-big-can-a-user-agent-string-get)

Reply:
To add to this, there's also the intention that [UA Client Hints](https://developer.mozilla.org/en-US/docs/Web/API/User-Agent_Client_Hints_API) will eventually replace UA strings generally speaking. How true this is, or how quickly it will happen is unsure, but I think this will provide value at the very least for now.

I do agree that UA strings should be avoided for using them for sniffing or feature detection, but I think for the purposes of audit logging, it's worth keeping track of these, provided that you opt-in in the bundle configuration. I think it'd be great to spot the difference between a user being on their phone vs a desktop or at least identifying it's a different device. Or in the context of data being modified, was it triggered by a web crawler/bot?

Doing some looking on my end as well, I think I'd like to continue with using the text type here, but keep in the truncating on the php side and have the limit configurable (I'll leave a comment there with details).

I'm not a huge fan of another relation just for this, especially with so many builds, oses, cpu architectures, etc, the number of entries will start to increase not only with new builds but device details. At the very least, I think we should take this approach for v1 and if it turns out to be a problem, we can normalize with a new release (and include a command to migrate the data).

Reply:
Sound like a reasonable middle ground, I'm in agreement that another relation is probably overkill at this stage, and certainly out of scope for this PR. Switching to text type with a configurable char limit seems reasonable.I'll get on this.